src/dns.rs

```diff
@@ -12,19 +15,27 @@ use crate::ip::IpVersionAddrIter;
 /// Ipv6 simulations use the link local subnet fe80:::/64
 pub struct Dns {
     addrs: IpVersionAddrIter,
-    names: IndexMap<String, IpAddr>,
+    mapping: IndexMap<String, NodeInfo>,
+}
+
+pub struct NodeInfo {
+    addrs: Vec<ScopedIpAddr>,
+    // id: NodeIdentifer
 }
 
 /// Converts or resolves to an [`IpAddr`].
 pub trait ToIpAddr: sealed::Sealed {
     #[doc(hidden)]
-    fn to_ip_addr(&self, dns: &mut Dns) -> IpAddr;
+    fn to_ip_addr(&self, dns: &Dns) -> Option<IpAddr>;
+
+    #[doc(hidden)]
+    fn to_name(&self, dns: &Dns) -> String;
```

Comment:
This part is only temporarily necessary.

The problem is, that we need a name to create a dns mapping. However not all nodes have a name using the 
current API since `Sim::client` expects a `impl ToIpAddr`. So to get a name for the dns registration,
this trait must temporarily be able to create such a name.

In the future all nodes should have explicit names, independent of their bound addresses, so this will only be temporary. Explicit names could be archived with the proposed builder API for node creation.

Reply:
I see, so you want to avoid a breaking change in this first change? If so, can you comment that here saying we'll remove this.

I'm inclined to just break things and work in a remote branch if that allows for less churn.